### PR TITLE
[AIRFLOW-6780] Add option to use args instead of command in K8s pod spec

### PR DESCRIPTION
**Pull request by brandonwillard**
This PR adds a config option, `kubernetes.worker_pod_uses_args`, that is passed to `PodGenerator.construct_pod` in order to decide whether or not the Kubernetes `args` field is used instead of `command` in the worker pod specs.

---
Issue link: [AIRFLOW-6780](https://issues.apache.org/jira/browse/AIRFLOW-6780)

Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Commit message/PR title starts with `[AIRFLOW-NNNN]`. AIRFLOW-NNNN = JIRA ID<sup>*</sup>
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

<sup>*</sup> For document-only changes commit message can start with `[AIRFLOW-XXXX]`.

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

**Comment by brandonwillard** [comment]
@dimberman, do those Travis failures have anything to do with this change?  It looks like something's timing out.

I'm seeing the same errors in https://github.com/apache/airflow/pull/7401, as well.

**Comment by mik-laj** [comment]
Travis is sad. I restarted the failed build, but it seems to me that this could be caused by a change in this PR.

File: airflow/kubernetes/pod_generator.py [hunk 1]
```diff
@@ -397,24 +397,32 @@ def reconcile_containers(base_containers: List[k8s.V1Container],
...
             - airflow.cfg
             - executor_config
             - dynamic arguments
         """
+
+        if use_args:
+            args = command
+            command = []
+        else:
+            args = []
```
Comment (mik-laj) [inline airflow/kubernetes/pod_generator.py:424]: It seems to me useful only for the Executor. What do you think about the constructor_pod method taking parameters args and command, and the main logic being in Kubernetes Executor?

Comment (brandonwillard) [reply airflow/kubernetes/pod_generator.py:424]: This functionality was only intended for the Executor, and, yeah, the question about where to put it did come to mind (i.e. between `PodGenerator` or `KubernetesExecutor`).

"Thematically" speaking, the option to choose the `args` or `command` field seemed most appropriate for `PodGenerator.construct_pod`, i.e. a method intended to generally construct pods, and one that already takes a `command` argument.  (Side note: I'm admittedly not a fan of the `use_args` "switch", but I didn't want to change an interface used by others, so I opted to simply augment/extend it.)

In general, I didn't want to add pod construction/manipulation logic to `KubernetesExecutor.run_next`; it just didn't seem good to have that kind of stuff spread around the code base.

**Comment by brandonwillard** [comment]
Sorry, I have to move this to my personal fork: #8146

**Pull request closed without merging**